### PR TITLE
Remove duplicate shuffle utilities

### PR DESCRIPTION
**Pull request by etiotto**
This PR contains the following changes:
  - remove duplicate shuffle functions and match NVidia implementation more closely
  - restructure Intel's `Targetinfo.cpp` and `Utility.cpp` to match NVidia implementation more closely
  - remove `toGenShuffleMode`
  - change Intel namespace to intel (lowercase)
  - add some copyright headers

### CODE REVIEW

File: third_party/intel/lib/TritonIntelGPUToLLVM/Utility.cpp [hunk 1]
```diff
@@ -29,63 +37,34 @@ Value loadShared(ConversionPatternRewriter &rewriter, Location loc, Value ptr,
...
-    val0 = commonShflSync(loc, rewriter, val0, i, mode, clamp);
-    val1 = commonShflSync(loc, rewriter, val1, i, mode, clamp);
-    vec = undef(vecTy);
-    vec = insert_element(vecTy, vec, val0, i32_val(0));
-    vec = insert_element(vecTy, vec, val1, i32_val(1));
-    return bitcast(vec, val.getType());
-  }
+static Value shuffleCommon(Location loc, ConversionPatternRewriter &rewriter,
+                           Value val, Value i, TritonGEN::ShflKind mode,
+                           Value clamp) {
```
Comment (etiotto) [inline third_party/intel/lib/TritonIntelGPUToLLVM/Utility.cpp:42]: Remove `clamp` as it is unused.

Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/Utility.cpp]: done

File: third_party/intel/lib/TritonIntelGPUToLLVM/HistogramOpToLLVM.cpp [hunk 2]
```diff
@@ -1,7 +1,7 @@
 #include "PatternTritonGPUOpToLLVM.h"
 #include "Utility.h"
-#include "mlir/Dialect/LLVMIR/NVVMDialect.h"
-#include "triton/Analysis/Utility.h"
+// #include "mlir/Dialect/LLVMIR/NVVMDialect.h"
+// #include "triton/Analysis/Utility.h"
```
Comment (whitneywhtsang) [inline third_party/intel/lib/TritonIntelGPUToLLVM/HistogramOpToLLVM.cpp:4]:
```suggestion
```

**Review by whitneywhtsang — APPROVED** [review]
Why is it preferred to rename Intel namespace to intel?

**Review by prathams417 — APPROVED** [review]

**Comment by etiotto** [comment]
> Why is it preferred to rename Intel namespace to intel?
`TargetInfo.cpp` used `mlir::triton::intel` while other files used `mlir::triton::Intel` so we were not consistent to start with.  I prefer the lowercase form because is easier to remember.

File: third_party/intel/lib/TritonIntelGPUToLLVM/HistogramOpToLLVM.cpp (hunk 2 above)
Comment (etiotto) [reply third_party/intel/lib/TritonIntelGPUToLLVM/HistogramOpToLLVM.cpp]: done

**Pull request merged**